libraries/entities/src/EntityTree.cpp

```diff
@@ -2240,8 +2249,19 @@ void EntityTree::fixupNeedsParentFixups() {
     }
 }
 
-void EntityTree::deleteDescendantsOfAvatar(QUuid avatarID) {
-    QHash<QUuid, QSet<EntityItemID>>::const_iterator itr = _childrenOfAvatars.constFind(avatarID);
+void EntityTree::knowAvatarID(const QUuid& avatarID) {
+    std::lock_guard<std::mutex> lock(_avatarIDsLock);
+    _avatarIDs += avatarID;
+}
+
+void EntityTree::forgetAvatarID(const QUuid& avatarID) {
+    std::lock_guard<std::mutex> lock(_avatarIDsLock);
+    _avatarIDs -= avatarID;
+}
+
+void EntityTree::deleteDescendantsOfAvatar(const QUuid& avatarID) {
+    std::lock_guard<std::mutex> lock(_childrenOfAvatarsLock);
+    auto itr = _childrenOfAvatars.find(avatarID);
     if (itr != _childrenOfAvatars.end()) {
```

Comment:
ok I think this was the real crash.  I assume `constFind` should check `!= constEnd()`.  In the old code, `constFind()` would NEVER equal `end()`, so it would always fall into this code and then crash.

Reply:
I wonder how this problem manifested for Darling but seemingly not for anyone else? (I couldn't repro the crash.)

Reply:
I agree, it’s weird!  could be a platform specific thing, maybe on some systems end() does equal constEnd()